modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -785,11 +787,17 @@ public Message getSampleData() throws SampleDataException {
 
   protected <R> MetadataResult<R> runWithMetadataContext(Function<MetadataContext, MetadataResult<R>> contextConsumer)
       throws MetadataResolvingException, ConnectionException {
+    return runWithMetadataContext(contextConsumer, metadataCache -> () -> getMetadataContext(metadataCache));
+  }
+
+  protected <R> MetadataResult<R> runWithMetadataContext(Function<MetadataContext, MetadataResult<R>> contextConsumer,
+                                                         Function<MetadataCache, Callable<MetadataContext>> metadataContextSupplier)
```

Comment:
why does this need to be a function and why is the cache the input?That doesn't sound very natural. Also, the IDEs get to supply their own Cache implementation so this is probably not super robust

Reply:
This was just to avoid having to duplicate the code or fill with optional contexts here... If you think this last one sounds more natural, I have no bigger concern

Reply:
not following. How does these functions provide what you mention? It's the modeling of the signatures what's weird to me. Can you clarify further?